cli/src/telemetry/session.rs

```diff
@@ -79,7 +91,14 @@ impl Session {
         }
 
         let url = format!("{}/telemetry", domain());
-        let body = serde_json::to_string(&self).unwrap();
+        let telemetry_session = serde_json::json!({
```

Comment:
This seems like an anti-pattern, we have a Telemetry struct, we should be able to serialize it with `serde_json`.

(I found this: https://serde.rs/attr-skip-serializing.html)

The thing that pops out first is that we have to manually write this out because `machine_id` is out of session in this PR, why is that? 

duplicating strings in maps and variable names that should match is a common source of bugs, and there is tooling to avoid it. 

Another options that is somewhat cumbersome but hard codes less strings is using `serde_json::to_value` which returns a [Option<Value>](https://docs.serde.rs/serde_json/value/enum.Value.html)

Reply:
Yeah, so this is funky and probably a sign that we should refactor a little more heavily. The reason we removed `machine_id` is because we felt that adding the `config` and `config_path` to the `session` (which is passed down to all commands) was a useful change which could be welcomed by all commands in the future.

The notion that `session` was _purely_ for `telemetry` didn't come across as it's something passed down everywhere. As such, it seemed like the natural place to add the new _session_ values.

Once we added these however, we had duplicate data in the `machine_id` (since it's really just stored in the `config`), and we didn't need to serialize `config` or `config_path` over to the Typescript `telemetry` logic.

I agree that a common source of bugs is mapping names in this manner. Maybe a better solution would be to introduce the notion of a `Session` which has less to do with `telemetry` (lives higher up in the file tree) and can send `Telemetry` messages (re-introducing the original struct in `Telemetry`) which it constructs in some well typed code?

Reply:
I'm new to this team and OSS in general, so as a result I feel like it's hard for me to advise on the best course of action here.
On the one hand, that refactor is not too related to this PR, and for the sake of progress I'd like to say it's ok to defer to a quick follow up that will focus only on refactoring as per your suggestion.
On the other hand, there's no clear specific owner for this currently, so I don't know who's priority it's going to be to do it? 
If Gravity is going to be adding more commands, it'd be great if we can settle the underlying structures to be clean and easy to build on top of. As early contributors, you're hitting these issues :)

So, from my perspective, I think I'm ok with a TODO FIXME (we don't fail the build on those) left in the code to refactor the serialization. We can discuss later who will own that. Sound good?

Reply:
That sounds like an excellent idea, and yeah, I also want to say that I haven't taken any of your comments to be grumbling or expecting us to magically know the direction to go in, but rather, "this ish is new...so probably this?"